src/modules/languages/rust.nix

```diff
@@ -13,67 +10,91 @@ let
           nixpkgs:
             follows: nixpkgs
   '';
+
+  error = dbg: "To use languages.rust.${dbg}, you need to add the following to your devenv.yaml:\n\n${setup}";
 in
 {
   options.languages.rust = {
     enable = lib.mkEnableOption "tools for Rust development";
 
-    packages = lib.mkOption {
+    components = lib.mkOption {
+      type = lib.types.listOf lib.types.str;
+      default = [ "rustc" "cargo" "clippy" "rustfmt" "rust-analyzer" ];
+      defaultText = lib.literalExpression ''[ "rustc" "cargo" "clippy" "rustfmt" "rust-analyzer" ]'';
+      description = ''
+        List of [Rustup components](https://rust-lang.github.io/rustup/concepts/components.html)
+        to install. Defaults to those available in `nixpkgs`.
+      '';
+    };
+
+    channel = lib.mkOption {
+      type = lib.types.enum [ "nixpkgs" "stable" "beta" "nightly" ];
```

Comment:
I just realized you need to set this to a non-nixpkgs value so you can set `toolchain`, which is kind of awkward and redundant. Not sure if this is something we can solve with the current API though, as it's kind of circular.